gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeatureConverter.java

```diff
@@ -80,9 +81,11 @@ private static PolygonFeature toFeatureFromGeometry(
     }
     ImmutableList.Builder<Point> vertices = ImmutableList.builder();
     for (Object point : (Object[]) coordinates) {
-      if(point instanceof GeoPoint){
+      if (point instanceof GeoPoint) {
         vertices.add(Point.newBuilder().setLongitude(((GeoPoint) point).getLongitude()).setLatitude(
             ((GeoPoint) point).getLatitude()).build());
+      } else {
+        Timber.d("Ignoring illegal point type in feature %s", doc.getId());
```

Comment:
Please prefer short circuit evaluation as in my example to prevent excessive nesting (i.e. `if (cond) return/break/continue;`) as code blocks grow over time.